tools/src/bin/sst-metadata.rs

```diff
@@ -91,6 +123,52 @@ async fn run(args: Args) -> Result<()> {
         metas.push(meta);
     }
 
+    if args.stats {
+        let mut file_stats = FileStatistics::default();
+        let mut field_stats_map = HashMap::new();
+        for (object_meta, sst_metadata, metadata_size, kv_size) in metas {
+            let parquet_meta = sst_metadata.parquet();
+
+            file_stats.file_count += 1;
+            file_stats.size += object_meta.size;
+            file_stats.metadata_size += metadata_size;
+            file_stats.kv_size += kv_size;
+            let filter_size = sst_metadata
+                .custom()
+                .parquet_filter
+                .as_ref()
+                .map(|f| f.size())
+                .unwrap_or(0);
+            file_stats.filter_size += filter_size;
+            file_stats.row_num += parquet_meta.file_metadata().num_rows();
+
+            let fields = parquet_meta.file_metadata().schema().get_fields();
+            for (_, row_group) in parquet_meta.row_groups().iter().enumerate() {
+                for i in 0..fields.len() {
+                    let column_meta = row_group.column(i);
+                    let field_name = fields.get(i).unwrap().get_basic_info().name().to_string();
+                    if !field_stats_map.contains_key(&field_name) {
+                        field_stats_map.insert(field_name.clone(), FieldStatistics::default());
+                    }
+                    let field_stats = field_stats_map.get_mut(&field_name).unwrap();
+                    field_stats.compressed_size += column_meta.compressed_size();
+                    field_stats.uncompressed_size += column_meta.uncompressed_size();
+                }
+            }
+        }
+        println!("{}", file_stats.to_string());
+
+        println!("FieldStatistics: ");
+        for (k, v) in field_stats_map.iter() {
+            println!("{},\t compressed_size: {:.2}mb,\t uncompressed_size: {:.2}mb,\t compress_ratio: {:.2}",
+                     k,
+                     as_mb(v.compressed_size as usize),
+                     as_mb(v.uncompressed_size as usize),
+                     v.uncompressed_size as f64 / v.compressed_size as f64);
```

Comment:
How to explain this?

From docs, compression_size represent pages with compression, and uncompressed_size represent page without uncompression, it seems they have no relation at all.

Reply:
uncompressed_size is the size before compress, compressed_size is the size after compress.